### PR TITLE
Updated Python Op and Image Pre-Processing Pipeline tutorials && Added CIFAR-10 Part 1 tutorial

### PR DESCRIPTION
**Pull request by MatthewInkawhich**
### Updated Image Pre-Processing Pipeline tutorial:

- Added __future__ imports for Python 3 support
- Minor markdown fixes/corrections
- Minor code fixes/adjustments (improved plot figure layout, changed print statements for Python 3)
- Removed portion of markdown relating to helpful links, as links were not set

### Updated Python Op tutorial

- Added __future__ imports for Python 3 support
- Markdown improvements and added code comments

### Added new CIFAR-10 Part 1 tutorial

- First part of 2-part tutorial (Part 2 is in progress)
- Gives detailed tutorial of training a model using a "custom" dataset of .png images

Changes authorized by @orionr and @janelu1 

Thanks to @harderthan 's [pull request](https://github.com/caffe2/caffe2/pull/1986), an additional typo fix was made on the Image Pre-Processing Pipeline tutorial.

### CODE REVIEW

**Comment by MatthewInkawhich** [comment]
@caffe2bot test this please

**Comment by orionr** [comment]
@caffe2bot test this please

**Comment by MatthewInkawhich** [comment]
@caffe2bot test this please

**Comment by orionr** [comment]
@caffe2bot retest this please

File: caffe2/python/tutorials/py_gen/CIFAR10_Part1.py [hunk 1]
```diff
@@ -0,0 +1,651 @@
...
+print("classes:", classes)
+
+
+# Now that we have our `classes` dictionary to map string labels to integer labels, we can write our label files for training, validation, and testing. We will split the data as follows:
+# 
+#     - training: 44,000 images (73%)
+#     - validation: 6,000 images (10%)
+#     - testing: 10,000 images (17%)
+#     
+# Note that the validation images are simply a subset of our training images that we will withhold to periodically test with during training. We do this so we can see how well our network is doing on unseen images without exposing our testing images to the model during training; something that makes machine learning experts cringe.
```
Comment (orionr) [inline caffe2/python/tutorials/py_gen/CIFAR10_Part1.py:187]: Nice line here. :)

File: caffe2/python/tutorials/py_gen/CIFAR10_Part1.py [hunk 2]
```diff
@@ -0,0 +1,651 @@
...
+# 
+# We will use the `update_dims` function as a helper to keep track of the dimensionality shrinkage that the convolutional and pooling layers cause. The dimensionality changes are as follows:
+# 
+# $$height_{out}=\frac{height_{in}-kernel+2*pad}{stride}+1$$
+# 
+# ---
+# 
+# $$width_{out}=\frac{width_{in}-kernel+2*pad}{stride}+1$$
+# 
+# While this function is not necessary, we found that it is a easy strategy to avoid having to hand calculate the dimensionality changes of the data to provide to the penultimate fully connected layer. It also allows us to quickly change hyperparameters such as kernel size and not have to worry about the corresponding dimensionality alterations.
```
Comment (orionr) [inline caffe2/python/tutorials/py_gen/CIFAR10_Part1.py]: Nit: it is a easy strategy -> it is an easy strategy

**Review by orionr — APPROVED** [review]
Looks great! Let's make sure the tests pass and then feel free to merge this.

**Comment by orionr** [comment]
Looks like we have some changes to the tutorials that need to be synced

```
15:22:47 ('tutorials_folder: ', '/var/lib/jenkins/workspace/caffe2/python/tutorials')
15:22:47 + git status
15:22:47 On branch merge-target
15:22:47 Changes not staged for commit:
15:22:47   (use "git add <file>..." to update what will be committed)
15:22:47   (use "git checkout -- <file>..." to discard changes in working directory)
15:22:47 
15:22:47 	modified:   py_gen/Python_Op.py
15:22:47 
15:22:47 Untracked files:
15:22:47   (use "git add <file>..." to include in what will be committed)
15:22:47 
15:22:47 	../../../env
15:22:47 
15:22:47 no changes added to commit (use "git add" and/or "git commit -a")
15:22:47 + git diff --quiet HEAD
15:22:47 After running a tutorial -> script sync there are changes. This probably means you edited an ipython notebook without a proper sync to a script. Please see caffe2/python/tutorials/README.md for more information
15:22:47 + echo 'After running a tutorial -> script sync there are changes. This probably means you edited an ipython notebook without a proper sync to a script. Please see caffe2/python/tutorials/README.md for more information'
```
After making final changes, run that script one more time on each of these? Thanks.

**Comment by MatthewInkawhich** [comment]
Made nits correction and re-synced Python_Op tutorial with its py_gen script.

**Comment by orionr** [comment]
@caffe2bot retest this please

**Pull request merged**